src/r_segs.c

```diff
@@ -157,7 +157,7 @@ void R_RenderMaskedSegRange(drawseg_t *ds, int x1, int x2)
       {
         if (!fixedcolormap)      // calculate lighting
           {                             // killough 11/98:
-            unsigned index = spryscale>>(LIGHTSCALESHIFT+hires);
+            unsigned index = spryscale >> (LIGHTSCALESHIFT + (video.xscale >> FRACBITS));
```

Comment:
> Also, walls appear darker in this PR compared to the last release (both with High Resolution off), but flats seem to be unaffected:

This is one of the places that I'm not sure about.

Reply:
Try to divide `spryscale` by the vertical scaling factor prior to `>> LIGHTSCALESHIFT`.

https://github.com/fabiangreffrath/woof/pull/1292/files#diff-4fcc8d4da7773102015f5441efb6d009a3d1fb277a6463c751f4bf199109d7c0R160

Reply:
Thanks, this helps: https://github.com/fabiangreffrath/woof/pull/1298/commits/6fbc2a1594e4dc3d880ceb4079a70c5c2b233a1b Still not perfect I think.

Reply:
It will be perfect for integer scale. Maybe use fixed point multiplication instead?

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/1298/commits/4243e3a2e2846d5539b032b9c1f0fac2a799902a